### PR TITLE
Ft linear constraints

### PR DESCRIPTION
**Pull request by LuziaKn**
Adds the possibility to represent obstacles through linear constraints in the mpc formulation.
Furthermore, multiple objective classes are now available which can be combined as desired.

### CODE REVIEW

**Comment by maxspahn** [comment]
@LuziaKn,

I looked through the code you have changed to make free-space-decomposition(fsd) available and made it work.
Specifically, I moved the computation of the fsd into the example using the class that was copied from the urdfenvs. The reason behind that is, that the mpcPlanner should not be polluted with sensor pre-processing. Ideally, there would be an optional pre-processing tool, but the planner should not have an attribute fsd by default, despite not needing it in most cases.
In the process, I moved all the helpers/utils into one folder. This folder should ideally be outside of models, but in the top-folder of robotmpcs in the future, as it has not only model-specific utils anymore.
It remains a problem that the optimization problem is often infeasible (errorflag=-7), and it takes two trys to cincrumnavigate the obstacle. Maybe, the constraints should be different over the time-horizon.

**Comment by LuziaKn** [comment]
The boxer now reaches the goal and the planner is feasible. The main issue was that at the first time step, the obstacle was not yet observed by the lidar due to how the observations were reset. 
I added a quick fix that updates the observation after the reset.

**Comment by maxspahn** [comment]
Now, it looks good to me.

File: examples/boxer_example.py [hunk 1]
```diff
@@ -71,39 +102,119 @@ def initialize_environment(self):
...
+                else:
+                    ref_q = q
+                linear_constraints_j, halfplanes_j = self.compute_constraints(ref_q, point_cloud)
+                linear_constraints.append(linear_constraints_j)
+                halfplanes.append(halfplanes_j)
+
+            self._planner.setLinearConstraints(linear_constraints, r_body=self._r_body)
+            # ---END: Preprocessing for planner---
+            action, output, exitflag = self._planner.computeAction(q, qdot, vel)
+            # ---START: Visualizations---
```
Comment (maxspahn) [inline examples/boxer_example.py:206]: This could potentially be a pre-processing. Issue for later.

**Pull request merged**